integration-tests/smoke/ccip_test.go

```diff
@@ -168,105 +151,233 @@ func TestTokenTransfer(t *testing.T) {
 		}
 	}
 
-	// Add all lanes
+	// Add all lanes.
 	require.NoError(t, ccdeploy.AddLanesForAll(e, state))
-	// Need to keep track of the block number for each chain so that event subscription can be done from that block.
-	startBlocks := make(map[uint64]*uint64)
-	// Send a message from each chain to every other chain.
-	expectedSeqNum := make(map[uint64]uint64)
 
-	twoCoins := new(big.Int).Mul(big.NewInt(1e18), big.NewInt(2))
-	tx, err := srcToken.Mint(
-		e.Chains[tenv.HomeChainSel].DeployerKey,
-		e.Chains[tenv.HomeChainSel].DeployerKey.From,
-		new(big.Int).Mul(twoCoins, big.NewInt(10)),
-	)
-	require.NoError(t, err)
-	_, err = e.Chains[tenv.HomeChainSel].Confirm(tx)
-	require.NoError(t, err)
+	// Deploy and approve tokens.
+	srcToken1, dstToken1 := deployAndApproveTokens(t, tenv, state, "Token1")
+	srcToken2, dstToken2 := deployAndApproveTokens(t, tenv, state, "Token2")
+
+	// Test scenarios are defined here
+	scenarios := []struct {
+		name         string
+		srcChain     uint64
+		dstChain     uint64
+		tokenAmounts []router.ClientEVMTokenAmount
+		receiver     common.Address
+		data         []byte
+	}{
+		{
+			name:     "Send token to EOA",
+			srcChain: tenv.HomeChainSel,
+			dstChain: tenv.FeedChainSel,
```

Comment:
Recommendation: I've been explicitly creating `sourceChain` and `destChain` variables in the beginning of the test (after the `NewLocalDevEnvironmentWithDefaultPrice` call) and using those instead of HomeChainSel and FeedChainSel, would be good if we stayed consistent with this.

Reply:
Also far less confusing that using these selectors which could def be source/dest chains but have the additional property of being home/feed chains